From a7ba7cdaba65673c6fc8935f8a3f227553d3b354 Mon Sep 17 00:00:00 2001
From: Takanari Hayama <taki@igel.co.jp>
Date: Thu, 21 Aug 2014 09:45:10 +0900
Subject: [PATCH 07/22] v4l2-renderer: pass v4l2_bo_state to
 set_output_buffer().

We should pass not only DMABUF, but also stride of the output buffer.
---
 src/v4l2-renderer-device.h |    2 +-
 src/v4l2-renderer.c        |    2 +-
 src/vsp-renderer.c         |    7 +++----
 3 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/src/v4l2-renderer-device.h b/src/v4l2-renderer-device.h
index a47aa98..e2e730c 100644
--- a/src/v4l2-renderer-device.h
+++ b/src/v4l2-renderer-device.h
@@ -72,7 +72,7 @@ struct v4l2_device_interface {
 	struct v4l2_renderer_device *(*init)(struct media_device *media);
 
 	struct v4l2_renderer_output *(*create_output)(struct v4l2_renderer_device *dev, int width, int height);
-	void (*set_output_buffer)(struct v4l2_renderer_output *out, int dmafd); 
+	void (*set_output_buffer)(struct v4l2_renderer_output *out, struct v4l2_bo_state *bo);
 
 	struct v4l2_surface_state *(*create_surface)(struct v4l2_renderer_device *dev);
 	int (*attach_buffer)(struct v4l2_surface_state *vs);
diff --git a/src/v4l2-renderer.c b/src/v4l2-renderer.c
index 15043c0..ec6803f 100644
--- a/src/v4l2-renderer.c
+++ b/src/v4l2-renderer.c
@@ -870,7 +870,7 @@ v4l2_renderer_output_set_buffer(struct weston_output *output, struct v4l2_bo_sta
 	vo->stride = bo->stride;
 	vo->map = bo->map;
 
-	device_interface->set_output_buffer(vo->output, bo->dmafd);
+	device_interface->set_output_buffer(vo->output, bo);
 
 	return;
 }
diff --git a/src/vsp-renderer.c b/src/vsp-renderer.c
index 2669c4c..7673c89 100644
--- a/src/vsp-renderer.c
+++ b/src/vsp-renderer.c
@@ -505,8 +505,6 @@ vsp_create_output(struct v4l2_renderer_device *dev, int width, int height)
 	fmt->fmt.pix_mp.height = height;
 	fmt->fmt.pix_mp.pixelformat = V4L2_PIX_FMT_ABGR32;
 	fmt->fmt.pix_mp.num_planes = 1;
-	fmt->fmt.pix_mp.plane_fmt[0].bytesperline = width * 4; //FIXME
-	fmt->fmt.pix_mp.plane_fmt[0].sizeimage = 0;
 
 	return (struct v4l2_renderer_output*)outdev;
 }
@@ -809,11 +807,12 @@ vsp_comp_set_view(struct v4l2_renderer_device *dev, struct v4l2_surface_state *s
 }
 
 static void
-vsp_set_output_buffer(struct v4l2_renderer_output *out, int dmafd)
+vsp_set_output_buffer(struct v4l2_renderer_output *out, struct v4l2_bo_state *bo)
 {
 	struct vsp_renderer_output *output = (struct vsp_renderer_output*)out;
 	DBG("set output dmafd to %d\n", dmafd);
-	output->surface_state.base.planes[0].dmafd = dmafd;
+	output->surface_state.base.planes[0].dmafd = bo->dmafd;
+	output->surface_state.fmt.fmt.pix_mp.plane_fmt[0].bytesperline = bo->stride;
 }
 
 static uint32_t
-- 
1.7.9.5

